### ISSUE TITLE
treeshake bug: object spread side effects incorrectly dropped

### ISSUE THREAD

**Issue by kzc** [rollup/rollup#4093]
- Rollup Version: rollup v2.50.0
- Operating System (or Browser): n/a
- Node Version (if applicable): n/a

### Repro

```
$ cat object-spread-effects.js
let result = "FAIL";
const unused = ({
    ...{
        get prop() {
            result = "PASS";
        }
    }
});
console.log(result);
```

### Expected Behavior

```
$ cat object-spread-effects.js | node
PASS
```

### Actual Behavior

```
$ cat object-spread-effects.js | rollup --silent | node
FAIL
```
```
$ cat object-spread-effects.js | rollup --silent
let result = "FAIL";
console.log(result);
```

**Comment by lukastaegert** [comment]
Can you post the complete reproduction including how the property is accessed?

**Comment by lukastaegert** [comment]
Fix at #4113
